### DISCUSSION TITLE
Nanobind example with setup tools

### DISCUSSION THREAD

**Discussion by petrasvestartas** [wjakob/nanobind#514] [Q&A]
Hi,

I can install nanobind example in my python in non-editable mode:
`C:/Users/petras/.rhinocode/py39-rh8/python.exe -m pip install .`

Although, I cannot run editable install though with -e:
`C:/Users/petras/.rhinocode/py39-rh8/python.exe -m pip install -e .`

```bash
ERROR: File "setup.py" or "setup.cfg" not found. Directory cannot be installed in editable mode: C:\brg\2_code\wood_cpp_nano
(A "pyproject.toml" file was found, but editable mode currently requires a setuptools-based build.)
Locally I can do it by pip install -e . but not for rhino python editor. Nano use pyproject.toml not setup.py which is a more preferred way to do that nowadays.
```
Any idea why?

**Comment by ThoreWietzke** [comment]
Do you use scikit_build_core as a backend? It works fine for me and they also (experimentally) support editable builds -> https://scikit-build-core.readthedocs.io/en/latest/configuration.html#editable-installs

Maybe that helps you?

**Reply by petrasvestartas** [reply]
I use the nanobind default sample which is based on sckit. Editable works locally in my conda environment. But not for another python editor I am using.

Instead of pip install -e . Shall i try?
 pip install --no-build-isolation --config-settings=editable.rebuild=true -Cbuild-dir=build -ve.

I guess the other python editor I use has only setup tools though...

**Reply by ThoreWietzke** [reply]
Yes, i use this with pybind11 and it works perfectly. I don't see, why it shouldn't work with nanobind as scikit_build_core manages the editable and rebuild stuff.

I don't know what you mean with python editor?

**Reply by petrasvestartas** [reply]
It is from a commercial software... Rhino3d that has a python 3.9.10 editor. So I need to give python.exe path where to install.

I tried this but it did not work:
```bash

(wood) C:\brg\2_code\wood_cpp_nano>C:/Users/petras/.rhinocode/py39-rh8/python.exe -m pip install --no-build-isolation --config-settings=editable.rebuild=true -Cbuild-dir=build -ve.

Usage:
  C:\Users\petras\.rhinocode\py39-rh8\python.exe -m pip install [options] <requirement specifier> [package-index-options] ...   
  C:\Users\petras\.rhinocode\py39-rh8\python.exe -m pip install [options] -r <requirements file> [package-index-options] ...    
  C:\Users\petras\.rhinocode\py39-rh8\python.exe -m pip install [options] [-e] <vcs project url> ...
  C:\Users\petras\.rhinocode\py39-rh8\python.exe -m pip install [options] [-e] <local project path> ...
  C:\Users\petras\.rhinocode\py39-rh8\python.exe -m pip install [options] <archive url/path> ...

no such option: --config-settings
```

Could you specify exact code for editable installes?

**Reply by ThoreWietzke** [reply]
This option is from scikit_build_core, not nanobind. I'm afraid I can't help you with that. You can try asking at https://github.com/scikit-build/scikit-build-core

**Reply by petrasvestartas** [reply]
Thank you anyway:)
